### DISCUSSION TITLE
posts/sync-point

### DISCUSSION THREAD

**Discussion by giscus** [VOID001/VOID001.github.io#2] [Announcements]
# posts/sync-point

 在测试多线程应用程序的时候，我们往往会构造一些多线程并发执行的 test case 用来测试是否有 race condition 或者 unexpected behavior 出现。通常情况下，开发者可能会将多个 function 并发执行 N 次，以此来判断最终结果是否符合预期，这可能对于大多数情况是足够的...

https://void-shana.moe/posts/sync-point

**Comment by VOID001** [comment]
此方法的一些局限性
1. 历史回归case较难维护，当代码被修改的时候可能导致某些回归例子失效
2. 无法对一个类的多个实例进行条件竞争测试
3. sync point 是否会对内存序造成影响，以至于对实际运行的代码产生干扰不确定
